core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -121,15 +121,21 @@ void clear() {
         }
 
         @Override
-        protected void store(ConfidentialKey key, byte[] payload) throws IOException {
-            LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getDigestOf(Util.toHexString(payload)));
-            data.put(key.getId(), payload);
+        protected void store(ConfidentialKey key, byte[] payload) throws IllegalStateException {
+                byte[] payloadDigest = Util.getSHA256DigestOf(payload);
```

Comment:
in the previous code the digest of the key would only be called if `FINE` level was enabled in the logger.
this will now always be called, even when fine logging is not enabled.